biapol_utilities/data/_data.py

```diff
@@ -25,6 +26,122 @@ def blobs():
+            padding = ((padding_y//2, padding_y//2), (padding_x//2, padding_x//2))
+        elif ((padding_y % 2) == 0) & ((padding_x % 2) != 0):
+            padding = ((padding_y//2, padding_y//2), ((padding_x//2) + 1, padding_x//2))
+        elif ((padding_y % 2) != 0) & ((padding_x % 2) == 0):
+            padding = (((padding_y//2) + 1, padding_y//2), (padding_x//2, padding_x//2))
+        else:
+            padding = (((padding_y//2) + 1, padding_y//2), ((padding_x//2) + 1, padding_x//2))
+        return(np.pad(image, padding))
+
+    def _create_image(self):
+        # Create hourglass image
+        self.image = np.zeros((self.time, self.z, self.y, self.x),
+                              dtype='uint8')
+        lower_content_radius = 1
+        radius_step = 1
+        for t in range(self.time):
+            glass_radius = self.y // 2
+            upper_content_radius = self.z // 4
+            for k in range(self.z):
+                # Create hourglass structure
+                if ((k == 1) | (k == self.z - 1)):
+                    fill = True  # Bottom and top are filled
+                else:
+                    fill = False
+                if k != 0:
+                    #  Draw circular glass wall
+                    disk = self._create_disk(glass_radius, fill)
+                    disk = self._pad2shape(disk, self.shape)
+                    self.image[t, k] =  disk * self.glass_value
+                #  Continuously decrease glass radius while in upper part
+                if k < ((self.z // 2) - 1):
+                    glass_radius -= radius_step
+                elif k == ((self.z // 2) - 1):
+                    glass_radius = 1
+                elif k == (self.z // 2):
+                    glass_radius = radius_step
+                #  Continuously increase glass radius while in lower part
+                else:
+                    glass_radius += radius_step
+                #  Add upper content (from quarter to half image)
+                if ((k >= self.z // 4) & (k < self.z // 2)):
+                    disk = self._create_disk(upper_content_radius, True)
+                    disk = self._pad2shape(disk, self.shape)
+                    self.image[t, k] += disk * self.content_value
+                    upper_content_radius -= radius_step
+
+            #  Add lower content: arc droplet
+            if t == 0:
+                self.image[t,
+                            self.z // 2,
+                            self.y // 2,
+                            self.x // 2] = self.content_value
+            if t >= 1:
+                half_disk = self._create_disk(lower_content_radius, False)
+                half_disk[:, :half_disk.shape[1] // 2] = 0
+                half_disk = self._pad2shape(half_disk, self.shape)
+                half_disk *= self.content_value
+                self.image[t, ((self.z // 2) + 2*t)] += half_disk
+                lower_content_radius += 2*radius_step
+
+
+def hourglass():
```

Comment:
I think I would add the options to set the hourglass shape/size in this function definition. After all, this is the function that is made public so I think it would be nice to make the parameters of the hourglass available to the users.

Meaning, you could copy the lines where the optional arguments are set for the hourglass class above into the argument of this function and also put the docstring here.

Reply:
I agree. I re-designed how data is generated there in order to allow shape definition by the user.
It can now be fully defined by its radius, or by a specific shape (under certain inherent restrictions, check information in [this commit](https://github.com/BiAPoL/biapol-utilities/pull/34/commits/fead93b864a04ac2e133270ccfbc23ad1da45698).
Also time dimension is editable now.

Reply:
Up to this point, I had to stick with the x = y = z size condition in order for it to work. I think for now, this should be ok.

Reply:
>It can now be fully defined by its radius, or by a specific shape (under certain inherent restrictions, check information in this commit.

I think it doesn't work *yet*. Currently, only `def hourglass()` has been made publically available, whereas `class Hourglas(...)` is not available to the API. When I try calling, for instance
```Python
import biapol_utilities as biau
gh = biau.data.hourglas(hourglas_value=2)
TypeError: hourglass() got an unexpected keyword argument 'glass_value'
```

Reply:
I added the parameters to the function, it should work now.

Reply:
Works for me and I can see the docstring when I write `help(biau.data.hourglas)`. Very clever idea to simply return Hourglass().image 👍 

I would suggest to comment the docstring from the class definition above to the function definition below so that it gets properly documented when calling `help(biau.data.hourglas)` or by Sphinx.

It could look like:

```Python
def hourglass(*args, **kwargs):
   """
   Generate a radially symmetrical gray-level (8-bit) 3D+t image.

    Attributes
    -----------
    radius : int, optional
        Hourglass maximum radius value. Default value: 100. Minimum value: 1.
        It also defines image shape as (radius, 2*radius + 1, 2*radius + 1,
                                        2*radius + 1).
    glass_value: uint8, optional
        Voxel intensity value of the hourglass walls.
    content_value: uint8, optional
        Voxel intensity value of the hourglass contents.
    image_shape: 4-tuple, optional
        Shape of desired resulting image. x, y and z must be equal. Minimum
        value: (1,3,3,3).
    pad_edges: bool, optional
        If True, pads image with 1 voxel wide border. This increases each
        spatial axes size by 2. Default: False.
    """
    return Hourglass(*args, **kwargs).image

```